### PR TITLE
user can create account and log in

### PR DESCRIPTION
**Pull request by pykennedy**
Add Firebase SDK to allow users to create an account and login. Currently this process is hardcoded for debugging purposes, it will later take in user input. I also added sketches for activities to be implemented in the future.

### CODE REVIEW

File: app/src/main/java/com/kennedy/peter/bloquery/ui/activity/LoginActivity.java [hunk 1]
```diff
@@ -40,11 +41,21 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    public void onAuthenticated(AuthData authData) {
+                        Toast.makeText(LoginActivity.this, "UID: " + authData.getUid(), Toast.LENGTH_SHORT).show();
+                    }
+
+                    @Override
+                    public void onAuthenticationError(FirebaseError firebaseError) {
+                        Toast.makeText(LoginActivity.this, "Error: " + firebaseError, Toast.LENGTH_SHORT).show();
+                    }
+                };
+                firebaseManager.logIn("fakestuff420@gmail.com", "q1w2e3r4", handler);
```
Comment (Electryc) [inline app/src/main/java/com/kennedy/peter/bloquery/ui/activity/LoginActivity.java:55]: I assume this is now just for testing and it'll be replaced with the actual credentials that the user inputs.

File: app/src/main/res/layout/activity_login.xml [hunk 2]
```diff
@@ -18,18 +18,18 @@
...
+            android:id="@+id/login_email"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:layout_alignParentTop="true"
             android:imeOptions="actionDone"
-            android:inputType="text"
-            android:maxLength="20"
-            android:hint="@string/username_hint"/>
+            android:inputType="textEmailAddress"
+            android:maxLength="30"
```
Comment (Electryc) [inline app/src/main/res/layout/activity_login.xml]: Why would you restrict the max length of the email?

**Comment by Electryc** [comment]
Usually when working in a team is very advised to provide a description of your pull request so that the reviewer knows what to focus on. 

Moreover, in oder to offer even a better context, when there are UI changes to the code, is advisable to provide a screenshot of how the application would look like when merging the pull request.

The above helps a lot in the review process across the team. 

For this incipient pull request maybe the screenshot would not be needed, but some kind of description would help. Something like:

> This PR adds the Firebase SDK and implements a basic flow for both login and creating a user. At the moment login is hardcoded so that we can easily test the Firebase integration. However this will change in the following pull requests.

**Comment by pykennedy** [comment]
i restricted the email on the assumption that other services might limit
the address space available, should i just let them go on forever?

**Comment by pykennedy** [comment]
updated the pull request. take a look at the sketches i made for the rest
of the app. The 2 hard parts im expecting is getting that slider in the
middle to scale the 2 layouts when the user drags it, and to get the user
profile picture to overlap the menu bar in the style i have shown. I will
definitely need help implementing those 2 features.

**Pull request merged**